### ISSUE TITLE
Version 2.10 only supports JSON responses, ignores statusCode and content-type headers

### ISSUE THREAD

**Issue by endymion** [awslabs/aws-sam-cli#361]
### Setup

Use any simple Hello-World Node handler function that returns content-type: text/html

### Expected result

The content-type on the response should be text/html

### Actual result

The content type on the HTTP response is application/json

(The status code also appears to be ignored.)

### Background

I discovered an apparent bug in the latest version of SAM Local today when I attempted to upgrade to support Node 8.10 and async / await.  Through troubleshooting, narrowed the problem down specifically to SAM Local.  I have example code for you to use for debugging.

This project demonstrates my simple attempt to convert a normal callback-based Lambda handler function to an async / await handler: https://github.com/endymion/async-handler

The problem has nothing to do with async / await.  You can see the problem with the default Hello-World project that AWS CodeStar generates for a serverless Node project.

The original Hello World handler function code is here: https://github.com/endymion/async-handler/blob/master/index.js#L6-L15

You can see that function in production here: https://d0nv0xlitb.execute-api.us-east-1.amazonaws.com/Prod/getWithCallback  I will leave that running for as long as possible.

This is what happens when you run that same function through SAM Local:

```
ec2-user:~/environment (master) $ curl -v 'http://localhost:8080/getAsync'
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET /getAsync HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.53.1
> Accept: */*
> 
< HTTP/1.1 200 OK
< Content-Type: application/json
< Content-Type: text/html
< Date: Fri, 13 Apr 2018 23:03:58 GMT
< Transfer-Encoding: chunked
< 
<!DOCTYPE html>
<html lang="en">
...
```

Note the incorrect content type.  There actually are two different content-type headers.  The one that I set, and for some reason "application/json".  When I attempted to change the status code during troubleshooting, it always returned status code 200.

This is a major problem that makes SAM Local nearly useless for local development, and we have been forced to resort to using AWS Lambda for development.  Because we don't have a way to simulate Lambda + API Gateway in development with our Node 8.10 code.

**Comment by endymion** [comment]
After some more reading in the issues here, this issue might be a duplicate of #325.

**Comment by endymion** [comment]
I found a workaround that got me back to development.  Until somebody can fix SAM Local.

If you're having trouble doing development on a non-JSON response, then you might want to try one of the Google Chrome extensions that can modify the response headers.  I used one called [Requestly](https://chrome.google.com/webstore/detail/requestly-redirect-url-mo/mdnleldcmiljblolnjhpnblkcekpdkpa/related?hl=en).

I had to configure one modification setting in that extension for each Lambda function that I'm working on.  I added the partial URL path, then selected "response" headers, "modify", and I told it to set the "Content-Type" header to "text/html".

This isn't pretty.  But the alternative for me right now would be callback hell, instead of using async / await in Node 8.10.  So at least this keeps me moving forward.
